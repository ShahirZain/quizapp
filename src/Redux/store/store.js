import { configureStore } from "@reduxjs/toolkit";
import reducer   from "../slicers/combineSlicer";


const store = configureStore({
  reducer: reducer,
});

export default store;