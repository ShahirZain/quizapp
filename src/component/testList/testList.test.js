import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../Redux/store/store";
import TestList from "./testList";
import utilityClass from "./utils";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

afterEach(cleanup);

const tree = renderer
  .create(
    <Provider store={store}>
      {" "}
      <MemoryRouter>
        <TestList></TestList>
      </MemoryRouter>{" "}
    </Provider>
  )
  .toJSON();

it("renders correctly", () => {
  expect(tree).toMatchInlineSnapshot(`
    Array [
      " ",
      <div
        className="testToBeGiven"
      >
        <h1
          className="test-list-heading"
        >
          Given test list
        </h1>
        <div
          className="js-test"
          onClick={[Function]}
        />
        <input
          className="text-test"
          data-testid="text-box"
          id="text"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Please Enter test key"
          type="text"
        />
      </div>,
      " ",
    ]
  `);
});

it("renders snapshot", () => {
  expect(tree).toMatchSnapshot();
});

const { getByTestId } = render(
  <Provider store={store}>
    <TestList />
  </Provider>
);
const text = getByTestId("text-box");

test("mock default function when text-box is leave", () => {
    const mockFn = jest.fn().mockReturnValue(console.log("temp Func Called!"))
    utilityClass.utility = mockFn
    fireEvent.mouseOut(text)
    expect(mockFn).toHaveBeenCalledTimes(1)
});
