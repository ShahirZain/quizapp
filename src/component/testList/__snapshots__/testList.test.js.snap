// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
Array [
  " ",
  <div
    className="testToBeGiven"
  >
    <h1
      className="test-list-heading"
    >
      Given test list
    </h1>
    <div
      className="js-test"
      onClick={[Function]}
    />
    <input
      className="text-test"
      data-testid="text-box"
      id="text"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Please Enter test key"
      type="text"
    />
  </div>,
  " ",
]
`;
